datanode/broker/sqlstore_broker.go

```diff
@@ -130,33 +130,35 @@ func (b *SQLStoreBroker) waitForFirstBlock(ctx context.Context, errCh <-chan err
 	} else if errors.Is(err, sqlstore.ErrNoLastBlock) {
 		lastProcessedBlock = entities.Block{
 			VegaTime: time.Time{},
-			Height:   -1,
```

Comment:
the problem here is if its anything other than 0 and data node has no hsitory it won't work.  it will blow on the check below:  Handling non-0 first block should be a seperate PR IMO that should not hold up this, there are other implications.



`if beginBlock.BlockNr() > lastProcessedBlock.Height+1 {
					return nil, fmt.Errorf("block height on begin block, %d, is too high, the height of the last processed block is %d",
						beginBlock.BlockNr(), lastProcessedBlock.Height)
				}
`